lesson15/main.js

```diff
@@ -0,0 +1,202 @@
+    removeCircle();
+    backButton.classList.remove("hidden");
+  }
+}
+
+async function fetchRenderData(inputName, inputNumber) {
+  const responseData = await fetchData(url);
+  if (responseData) {
+    renderData(responseData);
+    console.log(inputName, inputNumber); //show inputted number
+  }
+}
+
+openButton.addEventListener("click", () => {
+  promptMessage.textContent = "入力後ボタンを押してください";
+  promptMessage.style.color = "black";
+  nameBox.value = "";
+  numberBox.value = "";
+  setTimeout(() => {
+    nameBox.focus();
+  }, 0);
+  modal.classList.remove("hidden");
+  mask.classList.remove("hidden");
+  openButton.classList.add("hidden");
+  fetchButton.setAttribute("disabled", "true");
+  nameLabel.textContent = "名前";
+  nameLabel.style.color = "black";
+  numberLabel.textContent = "数字";
+  numberLabel.style.color = "black";
+});
+
+fetchButton.addEventListener("click", () => {
+  if (fetchButton.hasAttribute("disabled")) {
+    return;
+  }
+  const inputName = nameBox.value;
+  const inputNumber = numberBox.value;
+  fetchRenderData(inputName, inputNumber);
+  modal.classList.add("hidden");
+  mask.classList.add("hidden");
+});
+
+function closeModal() {
+  modal.classList.add("hidden");
+  mask.classList.add("hidden");
+  openButton.classList.remove("hidden");
+}
+
+mask.addEventListener("click", () => {
+  closeModal();
+});
+
+closeButton.addEventListener("click", () => {
+  closeModal();
+});
+
+backButton.addEventListener("click", () => {
+  backButton.classList.add("hidden");
+  openButton.classList.remove("hidden");
+  if (errorMessage) {
+    errorMessage.remove();
+  }
+  while (ul.firstChild) {
+    ul.firstChild.remove();
+  }
+});
+
+function checkInput() {
+  if (checkInputName() && checkInputNumber()) {
+    validInput();
+  }
+}
+
+function checkInputName() {
+  const inputName = nameBox.value;
+  if (inputName === "") {
+    nameLabel.textContent = "名前";
+    nameLabel.style.color = "black";
+    return false;
+  } else if (!inputName.match(personName)) {
+    invalidInputName();
+    return false;
+  } else {
+    return true;
+  }
+}
+
+function checkInputNumber() {
+  const inputNumber = numberBox.value;
+  if (inputNumber === "") {
+    numberLabel.textContent = "数字";
+    numberLabel.style.color = "black";
+    return false;
+  } else if (!inputNumber.match(halfWidthDigits)) {
+    invalidInputNumber();
+    return false;
+  } else {
+    return true;
+  }
+}
+
+function invalidInputName() {
+  nameLabel.textContent = "名前を入力してください";
+  nameLabel.style.color = "red";
+  fetchButton.disabled = true;
+}
+
+function invalidInputNumber() {
+  numberLabel.textContent = "半角数値を入力ください";
+  numberLabel.style.color = "red";
+  fetchButton.disabled = true;
+}
+
+function validInput() {
+  promptMessage.textContent = "入力後ボタンを押してください";
+  promptMessage.style.color = "black";
+  fetchButton.disabled = false;
+}
+
+nameBox.addEventListener("input", checkInput);
+numberBox.addEventListener("input", checkInput);
+checkInput();

```

Comment:
```suggestion
nameBox.addEventListener("input", checkInput);
numberBox.addEventListener("input", checkInput);
```
Q.
Since checkInput() is performed on each input by event, isn't the last checkInput() unnecessary?

Reply:
Thank you haru-san :)
I removed the checkInput().
I'm sorry about I didn't make commit about that.